### PR TITLE
Refactor: src/screens/BlockUser from Jest to Vitest 

### LINKED ISSUE #2544: Refactor: `src/screens/BlockUser`  from Jest to Vitest
**Is your feature request related to a problem? Please describe.**
This file is currently tested using Jest. As part of our migration to Vitest, we need to refactor this file's test cases to align with Vitest's syntax and features.

**Acceptance Criteria**
1. Replace Jest-specific functions and mocks with Vitest equivalents
2. Ensure all tests in `src/screens/BlockUser` pass after migration using `npm run test:vitest`
3. Maintain the test coverage for the file as 100% after migration
4. Upload a video or photo for this specific file coverage is 100% in the PR description

**Additional context**

- Refer to the merged PR for the base Vitest configuration:
https://github.com/PalisadoesFoundation/talawa-admin/pull/2457/files#diff-9a5a35cf8d7f02b2bba90905714b762560579c4789e9eb4cebb4607980aa3d93
- Vitest documentation:
https://vitest.dev/

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by amitb0ra**
**What kind of change does this PR introduce?**

refactoring

**Issue Number:** 

Fixes #2544

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**

image

**If relevant, did you update the documentation?**

No

**Summary**

This file is currently tested using Jest. As part of our migration to Vitest, we need to refactor this file's test cases to align with Vitest's syntax and features.
Closes: #2544

**Does this PR introduce a breaking change?**

No

**Other information**

Add ""@testing-library/dom": "^10.4.0"," due to problem in "import { render, screen } from '@testing-library/react';"

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

- **New Features**
	- Added a new testing library, `@testing-library/dom`, to enhance testing capabilities.

- **Bug Fixes**
	- Improved test structure and clarity for user blocking and unblocking functionalities.
	- Updated navigation simulation in tests for better control over browser history.

- **Refactor**
	- Simplified test imports and removed unnecessary asynchronous constructs to streamline the testing process.
	- Reduced the number of tests to focus on core functionalities.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
     "@pdfme/generator": "^5.2.3",
     "@pdfme/schemas": "^5.1.6",
     "@reduxjs/toolkit": "^2.3.0",
+    "@testing-library/dom": "^10.4.0",
```
[the remark this answers is not in the document]
Comment (amitb0ra) [reply package.json]: okay

**Comment by palisadoes** [comment]
Please fix the failing tests

**Comment by amitb0ra** [comment]
> Please fix the failing tests

which failing tests?

**Comment by amitb0ra** [comment]
> Please fix the failing tests

which failing tests?

**Comment by palisadoes** [comment]
1. Please make sure coderabbit.ai approves your PR
1. Make sure all tests pass (except introspection)
1. Please make sure that the code coverage for your patch reaches 100%

**Comment by palisadoes** [comment]
Please fix the failing tests

**Comment by palisadoes** [comment]
1. Please fix the failing tests. 
    1. You may have to fix a test that is not related to this PR as in some rare cases we merge faulty test code. If this is the case, fix the test in this PR. Do not open another issue.
1. Please make sure that CodeRabbit.ai approves the PR.

**Comment by palisadoes** [comment]
Please fix:

1. Any failing tests (Click on the `details` link for more information)
1. Any conflicting files

Make sure CodeRabbit.ai approves your changes

**Comment by palisadoes** [comment]
Please ask the `#talawa-admin` slack channel for assistance on this PR.

**Comment by amitb0ra** [comment]
@palisadoes any update?

**Comment by amitb0ra** [comment]
@palisadoes any update?

**Comment by amitb0ra** [comment]
any update?

@palisadoes

File: src/screens/BlockUser/BlockUser.spec.tsx [hunk 2]
```diff
@@ -241,102 +240,30 @@ const MOCKS = [
     },
   },
 ];
-const MOCKS_EMPTY = [
-  {
-    request: {
-      query: ORGANIZATIONS_LIST,
-      variables: {
-        id: 'orgid',
-      },
```
Comment (palisadoes) [inline src/screens/BlockUser/BlockUser.spec.tsx:250]: Why was this removed

File: package.json [hunk 3]
```diff
@@ -120,6 +120,7 @@
     "@testing-library/jest-dom": "^6.6.3",
     "@testing-library/react": "^16.0.1",
     "@testing-library/user-event": "^12.1.10",
+    "@testing-library/dom": "^10.4.0",
```
Comment (palisadoes) [inline package.json:123]: Why was this added? It doesn't appear to be used. Please revert this change.

File: src/screens/BlockUser/BlockUser.spec.tsx [hunk 4]
```diff
@@ -479,178 +398,6 @@ describe('Testing Block/Unblock user screen', () => {
...
-            <I18nextProvider i18n={i18nForTest}>
-              <BlockUser />
-            </I18nextProvider>
-          </Provider>
-        </BrowserRouter>
-      </MockedProvider>,
-    );
-
-    const input = screen.getByPlaceholderText('Search By First Name');
-    await act(async () => {
```
Comment (palisadoes) [inline src/screens/BlockUser/BlockUser.spec.tsx:629]: Why was this removed?

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comments

File: package.json (hunk 3 above)
Comment (amitb0ra) [reply package.json:123]: without this module i am getting error: `Module '"@testing-library/react"' has no exported member 'screen'.`

File: src/screens/BlockUser/BlockUser.spec.tsx (hunk 4 above)
Comment (amitb0ra) [reply src/screens/BlockUser/BlockUser.spec.tsx:629]: not required

File: src/screens/BlockUser/BlockUser.spec.tsx (hunk 2 above)
Comment (amitb0ra) [reply src/screens/BlockUser/BlockUser.spec.tsx:250]: not required

**Comment by amitb0ra** [comment]
any update?

@palisadoes

**Review by palisadoes — APPROVED** [review]

**Pull request merged**